projects/filigran-website/components/display-all-icons.tsx

```diff
@@ -1,16 +1,48 @@
+'use client'
 import * as FiligranIcon from 'filigran-icon'
-import {Fragment} from 'react'
+import React, {Fragment,ReactElement,useState,useMemo} from 'react'
+import Popup from './icon-popup';
+import SearchBar from './icon-searchbar';
 
 export const DisplayAllIcons = () => {
+  const [selectedIcon, setSelectedIcon] = useState<ReactElement | null>(null);
+  const [selectedIconLabel, setSelectedIconLabel] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const handleIconClick = (Icon : React.ComponentType<any> ,icon : string) => {
+    setSelectedIcon(<Icon className="w-20 h-20"/>);
```

Comment:
I think you can have only one state instead of 3.
You can use icon as a string.
If there is any icon set you can consider that you need to "open the modal" and from the popup you can get the Icon from the string.
It will avoid to have 3 state for one and the same purpose (showing the icon)